src/vue/forms/CreateIssuanceForm.vue

```diff
@@ -0,0 +1,234 @@
+    </div>
+    <div class="app__form-row">
+      <div class="app__form-field">
+        <div class="issuance-form__amount">
+          <input-field
+            type="number"
+            v-model="form.amount"
+            @blur="touchField('form.amount')"
+            id="create-issuance-amount"
+            :label="'issuance.amount' | globalize"
+            :error-message="getFieldErrorMessage('form.amount')"
+          />
+          <div
+            v-if="availableTokensAmount"
+            class="issuance-form__amount-label"
+          >
+            <span>{{ availableTokensAmount.currency }}</span>
+          </div>
+        </div>
+        <div
+          v-if="availableTokensAmount"
+          class="issuance-form__available"
+        >
+          <span>
+            {{ 'issuance.available-for-issuance' | globalize }}
+            {{ availableTokensAmount | formatMoney }}
+          </span>
+        </div>
+      </div>
+    </div>
+    <div class="app__form-row">
+      <div class="app__form-field">
+        <input-field
+          v-model="form.email"
+          @blur="touchField('form.email')"
+          id="create-issuance-email"
+          :label="'issuance.email' | globalize"
+          :error-message="getFieldErrorMessage('form.email')"
+        />
+      </div>
+    </div>
+    <div class="app__form-row">
+      <div class="app__form-field">
+        <input-field
+          v-model="form.reference"
+          @blur="touchField('form.reference')"
+          id="create-issuance-reference"
+          :error-message="getFieldErrorMessage('form.reference')"
+          :label="'issuance.reference' | globalize"
+        />
+      </div>
+    </div>
+    <div class="app__form-actions">
+      <button
+        v-ripple
+        type="submit"
+        class="issuance-form__submit-btn"
+        :disabled="formMixin.isDisabled"
+      >
+        {{ 'issuance.issue' | globalize }}
+      </button>
+      <button
+        v-ripple
+        type="button"
+        class="issuance-form__cancel-btn"
+        :disabled="formMixin.isDisabled"
+        @click.prevent="closeForm"
+      >
+        {{ 'issuance.cancel' | globalize }}
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+import FormMixin from '@/vue/mixins/form.mixin'
+import SelectField from '@/vue/fields/SelectField'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { Sdk } from '@/sdk'
+import { base } from '@tokend/js-sdk'
+
+import { required, email, between } from '@validators'
+import { vuexTypes } from '@/vuex'
+import { mapGetters } from 'vuex'
+
+export default {
+  name: 'create-issuance-form',
+  components: {
+    SelectField
+  },
+  mixins: [FormMixin],
+  props: {
+    isShown: { type: Boolean, default: true }
+  },
+  data: _ => ({
+    form: {
+      asset: '',
+      amount: 0.00001,
+      email: '',
+      reference: ''
+    },
+    userOwnedTokens: null
+  }),
+  validations () {
```

Comment:
do we need it to be a function?

Reply:
yep it makes it dynamical, like `computed`